src/tools/interop/idt/capture/shell_utils.py

```diff
@@ -0,0 +1,81 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import shlex
+import subprocess
+
+from mobly.utils import stop_standing_subprocess
+
+
+class BashRunner:
+    """
+    Uses subprocess to execute bash commands
+    Intended to be instantiated and then only interacted with through instance methods
+    """
+
+    def __init__(self, command: str, sync: bool = False,
```

Comment:
I would personally avoid calling `Popen` functions that can raise an error from within the `__init__` function, but it's okay. You might want to document the exception that could be raised. I can imagine a lot of user errors that will result in exceptions being raised from this module.